tensorflow/python/ops/nn.py

```diff
@@ -801,6 +803,71 @@ def batch_norm_with_global_normalization(t,
   return batch_normalization(t, m, v, beta, gamma if scale_after_normalization
                              else None, variance_epsilon, name)
 
+def batch_norm_training_wih_rolling_moments(t, beta, gamma, variance_epsilon,
+                                            rolling_means, rolling_inv_var,
+                                            data_format="NCHW"):
+    """ Batch Normalization in training mode
+    Args:
+      t: A 4D input Tensor.
+      beta: A 1D beta Tenor with size matching the 'C' dimention of t.
+        An offset to be addded to the normalized tensor.
+      gamma: A 1D beta Tenor with size matching the 'C' dimention of t.
+        A scalar to be multipled with the normalized tensor
+      variance_epsilon: A small float number to avoid dividing by 0.
+      rolling_means: A 1D Variable to store rolling means.
+      rolling_inv_var: A 1D Variable to store rolling inverse variances.
+
+    Returns:
+      output: a batch-normalized `t`.
+      mean: 1D tensor with computed means for the batch.
+      inv_var: A 1D tensor with computed inverse variances of the batch.
+    """
+    if data_format != "NCHW":
+        raise AttributeError("Currently only NCHW data_format is supported.")
+    output, mean, inv_var = batch_normalize_training(t,
+        gamma, beta, running_mean, running_inv_var, 0.01, 0.8)
+    return output, mean, inv_var
+
+def batch_norm_inference(t, beta, gamma, variance_epsilon, population_means,
```

Comment:
Nope I have not. I am working under the assumption that Cudnn will be a lot faster but am lazy for now and want to have one less thing to test while still working on this. I will do the test some point later.